core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -885,31 +884,39 @@ impl<R: WriteableRocks> StateManagerDatabase<R> {
     fn ensure_historical_substate_values(&self) {
         let db_context = self.open_rw_context();
         let extension_data_cf = db_context.cf(ExtensionsDataCf);
-        let values_associated_since = extension_data_cf
-            .get(&ExtensionsDataKey::ValuesAssociatedWithStateHashTreeSince)
-            .map(|bytes| scrypto_decode::<StateVersion>(&bytes).unwrap());
+        let status = extension_data_cf
+            .get(&ExtensionsDataKey::StateHashTreeAssociatedValuesStatus)
+            .map(|bytes| {
+                scrypto_decode::<VersionedStateHashTreeAssociatedValuesStatus>(&bytes)
+                    .unwrap()
+                    .into_latest()
+            });
 
         if self.config.enable_historical_substate_values {
-            if let Some(version) = values_associated_since {
-                info!("Historical Substate values enabled since {:?}", version);
+            if let Some(status) = status {
+                info!("Historical Substate values enabled since {:?}", status);
             } else {
                 let current_version = self.max_state_version();
                 info!(
                     "Enabling historical Substate values at {:?}",
                     current_version
                 );
                 self.populate_hash_tree_associated_substate_values(current_version);
+                let status = StateHashTreeAssociatedValuesStatusV1 {
```

Comment:
Can we add a comment explaining why this is safe if this process is disturbed? (i.e. idempotent or something?)

Although technically it might leak values if it's disturbed half way through and disabled; because we've never saved the fact we're indexing?

That's probably okay. We can mention an improvement might be to store a status of "IndexingInProgress { from_inclusive: X, to_exclusive: Y }" to allow us to wipe / commence as appropriate.

Reply:
> Can we add a comment explaining why this is safe if this process is disturbed?

This comment is already there, at line 882 of this file (https://github.com/radixdlt/babylon-node/pull/872/files/059ba438cbd1dda448393ec3a6860c7988970c8e#diff-5cf22ad2b534dd277f51718bd8cf5c3e624cecbc4819906008c39e0798427ce3R882)

> Although technically it might leak values if it's disturbed half way through and disabled; because we've never saved the fact we're indexing?

Luckily, "disabling" will also wipe the entire history table, because [GC needs this behavior](https://github.com/radixdlt/babylon-node/pull/873/files#diff-5cf22ad2b534dd277f51718bd8cf5c3e624cecbc4819906008c39e0798427ce3R917). I think we are safe from leaks here.

Reply:
aaaaah I see, that's the leak!

In this case, it seems really low-cost to just do `.delete_all()` on every boot on which the state history is disabled. It is a single RocksDB "delete range" operation. Since I don't trust myself anymore - would it mean no leak?

Reply:
(addressed by https://github.com/radixdlt/babylon-node/pull/873/commits/a64e853f67b42854e6c9b1587d78424694a12a26 in the GC's PR)